istioctl/cmd/tag.go

```diff
@@ -276,63 +255,36 @@ revision tag before removing using the "istioctl x revision tag list" command.
 }
 
 // setTag creates or modifies a revision tag.
-func setTag(ctx context.Context, kubeClient kube.ExtendedClient, tag, revision string, generate bool, w, stderr io.Writer) error {
-	// abort if there exists a revision with the target tag name
-	revWebhookCollisions, err := getWebhooksWithRevision(ctx, kubeClient, tag)
-	if err != nil {
-		return err
-	}
-	if !generate && !overwrite && len(revWebhookCollisions) > 0 {
-		return fmt.Errorf("cannot create revision tag %q: found existing control plane revision with same name", tag)
-	}
-
-	// find canonical revision webhook to base our tag webhook off of
-	revWebhooks, err := getWebhooksWithRevision(ctx, kubeClient, revision)
+func setTag(ctx context.Context, kubeClient kube.ExtendedClient, tagName, revision string, generate bool, w, stderr io.Writer) error {
+	opts := &tag.GenerateOptions{
+		Tag:           tagName,
+		Revision:      revision,
+		WebhookName:   webhookName,
+		ManifestsPath: manifestsPath,
+		Generate:      generate,
+		Overwrite:     overwrite,
+	}
+	tagWhYAML, err := tag.Generate(ctx, kubeClient, opts)
 	if err != nil {
 		return err
 	}
-	if len(revWebhooks) == 0 {
-		return fmt.Errorf("cannot modify tag: cannot find MutatingWebhookConfiguration with revision %q", revision)
-	}
-	if len(revWebhooks) > 1 {
-		return fmt.Errorf("cannot modify tag: found multiple canonical webhooks with revision %q", revision)
-	}
 
-	whs, err := getWebhooksWithTag(ctx, kubeClient, tag)
-	if err != nil {
-		return err
-	}
-	if len(whs) > 0 && !overwrite {
-		return fmt.Errorf("revision tag %q already exists, and --overwrite is false", tag)
-	}
-
-	tagWhConfig, err := tagWebhookConfigFromCanonicalWebhook(revWebhooks[0], tag)
-	if err != nil {
-		return fmt.Errorf("failed to create tag webhook config: %v", err)
-	}
-	tagWhYAML, err := tagWebhookYAML(tagWhConfig, manifestsPath)
-	if err != nil {
-		return fmt.Errorf("failed to create tag webhook: %v", err)
-	}
-
-	// Check the newly generated webhook does not conflict with existing ones
+	// Check the newly generated webhook does not conflict with existing ones.
 	resName := webhookName
 	if resName == "" {
-		resName = fmt.Sprintf("%s-%s", "istio-revision-tag", tag)
+		resName = fmt.Sprintf("%s-%s", "istio-revision-tag", tagName)
 	}
 	if err := analyzeWebhook(resName, tagWhYAML, kubeClient.RESTConfig()); err != nil {
 		// if we have a conflict, we will fail. If --skip-confirmation is set, we will continue with a
 		// warning; when actually applying we will also confirm to ensure the user does not see the
 		// warning *after* it has applied
-		if skipConfirmation {
+		if !skipConfirmation {
```

Comment:
Not sure I understand why this was inverted?

Reply:
When we automatically set the default revision in the installer, we need it to happen without user confirmation (like when we do `install -y` in test framework). This makes it so that we only do the duplicate injection analysis and confirmation if `skip-confirmation` is disabled.